python/src/uagents/communication.py

```diff
@@ -163,10 +163,12 @@ async def send_exchange_envelope(
                     success = resp.status == 200
                     if success:
                         if sync:
-                            # If the message is synchronous but not verified, return the envelope
                             env = Envelope.model_validate(await resp.json())
-                            if env.signature is None:
-                                return env
+                            if env.signature and not env.verify():
+                                errors.append(
+                                    "Received response envelope that failed verification"
+                                )
+                                continue
```

Comment:
This is considered to be a failed delivery even though a response was received and is consistent with the case when an unsigned envelope is received from an agent via the asgi server: https://github.com/fetchai/uAgents/blob/main/python/src/uagents/asgi.py#L273-L289.